dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/InvokeTelnetHandler.java

```diff
@@ -147,76 +152,81 @@ public String telnet(Channel channel, String message) {
                 }
             }
         }
+
         if (i < 0 || !message.endsWith(")")) {
             return "Invalid parameters, format: service.method(args)";
         }
+
         String method = message.substring(0, i).trim();
         String args = message.substring(i + 1, message.length() - 1).trim();
         i = method.lastIndexOf(".");
         if (i >= 0) {
             service = method.substring(0, i).trim();
             method = method.substring(i + 1).trim();
         }
+
         List<Object> list;
         try {
             list = JSON.parseArray("[" + args + "]", Object.class);
         } catch (Throwable t) {
             return "Invalid json argument, cause: " + t.getMessage();
         }
-        if (paramClasses != null) {
-            if (paramClasses.length != list.size()) {
+        if (paramTypes != null) {
+            if (paramTypes.length != list.size()) {
                 return "Parameter's number does not match the number of parameter class";
             }
             List<Object> listOfActualClass = new ArrayList<>(list.size());
             for (int ii = 0; ii < list.size(); ii++) {
                 if (list.get(ii) instanceof JSONObject) {
                     JSONObject jsonObject = (JSONObject) list.get(ii);
-                    listOfActualClass.add(jsonObject.toJavaObject(paramClasses[ii]));
+                    listOfActualClass.add(jsonObject.toJavaObject(paramTypes[ii]));
                 } else {
                     listOfActualClass.add(list.get(ii));
                 }
             }
             list = listOfActualClass;
         }
-        Invoker<?> invoker = null;
+
         Method invokeMethod = null;
-        for (Exporter<?> exporter : DubboProtocol.getDubboProtocol().getExporters()) {
-            if (service == null || service.length() == 0) {
-                invokeMethod = findMethod(exporter, method, list, paramClasses);
-                if (invokeMethod != null) {
-                    invoker = exporter.getInvoker();
-                    break;
-                }
-            } else {
-                if (service.equals(exporter.getInvoker().getInterface().getSimpleName())
-                        || service.equals(exporter.getInvoker().getInterface().getName())
-                        || service.equals(exporter.getInvoker().getUrl().getPath())) {
-                    invokeMethod = findMethod(exporter, method, list, paramClasses);
-                    invoker = exporter.getInvoker();
-                    break;
-                }
+        ProviderModel selectedProvider = null;
+        for (ProviderModel provider : ApplicationModel.allProviderModels()) {
+            if (provider.getServiceName().equalsIgnoreCase(service)
```

Comment:
Should we refactor this to a another method something like 
if (isServiceMatch(provider,service) {
  -- existing code ---
}
public static boolean isServiceMatch( ProviderModel provider,String service) {
   return StringUtils.isEmpty(service) 
              ||provider.getServiceName().equalsIgnoreCase(service)
              ||provider.getServiceName().equalsIgnoreCase(service)
              ||provider.getServiceInterfaceClass().getSimpleName().equalsIgnoreCase(service)
              ||provider.getServiceInterfaceClass().getName().equalsIgnoreCase(service)
}

Reply:
done.